tensorflow/python/distribute/input_lib.py

```diff
@@ -1430,6 +1439,7 @@ def __init__(self, dataset, worker, devices):
     self._worker = worker
     self._devices = devices
     self._element_spec = dataset.element_spec
+    self._replication_mode = replication_mode
```

Comment:
This was added since the when dataset was not None in its derived class "_SingleWorkerOwnedDatasetIterator" it would fall back to base class constructor and (l.1569) and then _make_iterator(self) would have failed. I removed it for now and instead added "self._replication_mode = replication_mode" to before dataset checkout line 1585,